From 6ed6a2489bd83af09a6b824f91ae2856d9e5f8fe Mon Sep 17 00:00:00 2001
From: Geert Uytterhoeven <geert@linux-m68k.org>
Date: Tue, 18 Nov 2008 20:54:08 +0100
Subject: [PATCH 27/90] m68k: Disable Atari serial console support if modular

If CONFIG_ATARI_SCC=m, I get the following warnings:

| drivers/char/atari_scc.c: At top level:
| drivers/char/atari_scc.c:1573: warning: data definition has no type or storage class
| drivers/char/atari_scc.c:1573: warning: type defaults to 'int' in declaration of 'console_initcall'
| drivers/char/atari_scc.c:1573: warning: parameter names (without types) in function declaration
| drivers/char/atari_scc.c:1567: warning: 'atari_scc_console_init' defined but not used

Apparently console_initcall() is not defined in the modular case.

Disable serial console support if the driver is modular.

Signed-off-by: Geert Uytterhoeven <geert@linux-m68k.org>
---
 drivers/char/atari_scc.c |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/char/atari_scc.c b/drivers/char/atari_scc.c
index f1ade4a..800193c 100644
--- a/drivers/char/atari_scc.c
+++ b/drivers/char/atari_scc.c
@@ -1401,10 +1401,12 @@ static int scc_break_ctl(struct tty_struct *tty, int break_state)
 }
 
 
+#if defined(CONFIG_SERIAL_CONSOLE) && !defined(MODULE)
+
 /*---------------------------------------------------------------------------
  * Serial console stuff...
  *--------------------------------------------------------------------------*/
-#if 1
+
 #define scc_delay() \
 	asm volatile ("tstb %0" : : "m" (*scc_del) : "cc")
 
@@ -1571,7 +1573,8 @@ static int __init atari_scc_console_init(void)
 }
 
 console_initcall(atari_scc_console_init);
-#endif
+
+#endif /* CONFIG_SERIAL_CONSOLE && !MODULE */
 
 /***************************** End of Functions *********************/
 
-- 
1.6.2.3

